stats/opencensus/e2e_test.go

```diff
@@ -0,0 +1,773 @@
+	leakcheck.RegisterIgnoreGoroutine("go.opencensus.io/stats/view.(*worker).start")
+}
+
+var defaultTestTimeout = 5 * time.Second
+
+type fakeExporter struct {
+	t *testing.T
+
+	mu        sync.RWMutex
+	seenViews map[string]*viewInformation
+}
+
+// viewInformation is information Exported from the view package through
+// ExportView relevant to testing, i.e. a reasonably non flaky expectation of
+// desired emissions to Exporter.
+type viewInformation struct {
+	aggType    view.AggType
+	aggBuckets []float64
+	desc       string
+	tagKeys    []tag.Key
+	rows       []*view.Row
+}
+
+func (fe *fakeExporter) ExportView(vd *view.Data) {
+	fe.mu.Lock()
+	defer fe.mu.Unlock()
+	fe.seenViews[vd.View.Name] = &viewInformation{
+		aggType:    vd.View.Aggregation.Type,
+		aggBuckets: vd.View.Aggregation.Buckets,
+		desc:       vd.View.Description,
+		tagKeys:    vd.View.TagKeys,
+		rows:       vd.Rows,
+	}
+}
+
+func (vi *viewInformation) equal(vi2 *viewInformation) bool {
+	if vi == nil && vi2 == nil {
+		return true
+	}
+	if (vi != nil) != (vi2 != nil) {
+		return false
+	}
+	if vi.aggType != vi2.aggType {
+		return false
+	}
+	if !cmp.Equal(vi.aggBuckets, vi2.aggBuckets) {
+		return false
+	}
+	if vi.desc != vi2.desc {
+		return false
+	}
+	if !cmp.Equal(vi.tagKeys, vi2.tagKeys, cmp.Comparer(func(a tag.Key, b tag.Key) bool {
+		return a.Name() == b.Name()
+	})) {
+		return false
+	}
+	if !compareRows(vi.rows, vi2.rows) {
+		return false
+	}
+	return true
+}
+
+// compareRows compares rows with respect to the information desired to test.
+// Both the tags representing the rows and also the data of the row are tested
+// for equality. Rows are in nondeterministic order when ExportView is called,
+// but handled because ExportView is called every default metrics reporting
+// interval, which will work eventually in a reasonable manner (i.e. doesn't
+// flake).
+func compareRows(rows []*view.Row, rows2 []*view.Row) bool {
+	if rows == nil && rows2 == nil {
+		return true
+	}
+	if (rows != nil) != (rows2 != nil) {
+		return false
+	}
+	if len(rows) != len(rows2) {
+		return false
+	}
+
+	for i, row := range rows {
+		if !cmp.Equal(row.Tags, rows2[i].Tags, cmp.Comparer(func(a tag.Key, b tag.Key) bool {
+			return a.Name() == b.Name()
+		})) {
+			return false
+		}
+		if !compareData(row.Data, rows2[i].Data) {
+			return false
+		}
+	}
+	return true
+}
+
+// compareData returns whether the two aggregation data's are equal to each
+// other with respect to parts of the data desired for correct emission. The
+// function first makes sure the two types of aggregation data are the same, and
+// then checks the equality for the respective aggregation data type.
+func compareData(ad view.AggregationData, ad2 view.AggregationData) bool {
+	if ad == nil && ad2 == nil {
+		return true
+	}
+	if (ad != nil) != (ad2 != nil) {
+		return false
+	}
+	switch ad1 := ad.(type) {
+	case *view.DistributionData:
+		dd2, ok := ad2.(*view.DistributionData)
```

Comment:
If you want you can do:

```go
if reflect.TypeOf(ad) != reflect.TypeOf(ad1) {
	return false
}
```

And then assume the types match from there down.  (I.e. you don't even need cases for `SumData` or `LastValueData`.)